### ISSUE TITLE
Clean TweakScale's Module from unchanged parts on save

### ISSUE THREAD

**Issue by Lisias** [net-lisias-ksp/TweakScale#85]
That's the thing: not every challenge accepts TweakScale. Not a problem, just a fact.

But it's pretty annoying to have similar installments due this. So it would be nice if we could easily strip TweakScale modules from the unscaled parts while saving.

On Issue #20 I successfully intercepted the save event in order to rename the extra TweakScale rogue Module on a part. The idea is to further purse the concept by bluntly not saving the TweakScale module info when the current scale is equal to the default (or is 100%, depending of the scaletype). This way, unchanged parts will not have the TweakScale module on it - by side effect, a craft whose parts where not scale will not have TweakScale on any of them.

This is not a problem because TweakScale is injected back on every Load when it's present and no Module info is present on the parts.

Another nice feature would be a Configuration Panel to "hide" TweakScale from the UI when the user don't wanna be bored with them. Not sure if I will be able to implement this on this issue, but it worth to take note if it and try.

**Comment by Lisias** [comment]
This was partially implemented by #170 (the part where TweakScale UI is not shown when deactivated)

**Comment by Lisias** [comment]
Implemented on commit https://github.com/net-lisias-ksp/TweakScale/commit/783c2b47490389c3f72f7d700578d0ee858f83c6 .

A probably mostly harmless side effect was detected - I could not managed to trick KSP into just not saving the node, an node is always saved. So I just cleared the node from all data, and the following is saved on the craft file instead:

```
    MODULE
    {
        UPGRADESAPPLIED
        {
        }
    }
```

Subsequent saves **do not** have this clutter anymore, so KSP handles this somehow in a nice way.

**Comment by Lisias** [comment]
Tested down to KSP 1.4.4 , no collateral effected detected.

Since this stunt will not be applied on KSP < 1.40 (as they don't have the Upgrade Pipeline, and this would force the user to remove and reapply a new part to have TweakScale back), I didn't bored to test on them.

In a way or another, #170 implementes `active` and `available` that will allow forced deactivation of TweakScale anyway - it will just not be decluttered from the craft file on KSP 1.3.1 and previous.

**Comment by Lisias** [comment]
This is going to give PartModule a run for its money! :)

```
[WRN 11:31:46.340] [Part]: PartModule indexing mismatch at adapterSmallMiniShort, index 0.
Node 'ModulePartInfo' found in loaded data, but 'TweakScale' is defined in prefab.
Looking for ModulePartInfo in other indices...
[WRN 11:31:46.341] ...ModulePartInfo module found at index 1.
[LOG 11:31:46.343] [FlightGlobals]: Part persistentId changed from 3819051273 to 4032516425. Vessel PersistentId 0
[ERR 11:31:46.343] Cannot load PartModule #0 because ConfigNode contains no module name
```

Other than an ERR entry on the KSP.log, nothing else is happening - the user is not being pesked by a missing module neither, so from the user point of view, the feature works transparently.

**Comment by Lisias** [comment]
The undesired behaviour described in the last post was worked around on commit https://github.com/net-lisias-ksp/TweakScale/commit/1201c2829113b0be1dd2b580e383a811a626ed1d

```
     // Nothing to see here. Please ignore me.
    {
        UPGRADESAPPLIED
        {
        }
```

Now KSP.log doesn't logs an ERR entry anymore. Other than the Upgrade Pipeline working to sort its way out on the new module's indices, no collateral effect was detected at all. Nice!

**Comment by Lisias** [comment]
Closed, but postponed to 2.4.5.1 as I want some time with thins on the Beta program first.

Just in case...
